content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -19,6 +19,301 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
+
+DEMO PROFILE IMAGE
+
+As can be seen, this was unfortunately leaving a lot of performance on the table. Many ideas were discussed over a long period for how to improve this. The main two paths forward were:
+
+1. Persist render world entities and their component data across frames - this has the problem of memory leaks and Entity collisions
+2. Stop using entities for storing component data in the render world
+
+We have decided to explore option 2 for Bevy 0.12 as persisting entities involves solving other problems that have no simple and satisfactory answers.
+
+After consideration, we landed on using `HashMap<Entity, T>` with a hash function designed by @SkiFire13, and inspired by `rustc-hash`. This configuration is called `EntityHashMap` and is the new way to store component data in the render world.
+
+#### EntityHashMap Helpers
+
+A helper plugin was added to make it simple and quick to extract main world data for use in the render world in the form of `ExtractInstancesPlugin`. You can extract all entities matching a query, or only those that are visible, extracting multiple components at once into one target type.
+
+It is a good idea to group component data that will be accessed together into one target type to avoid having to do multiple lookups.
+
+To extract two components from visible entities:
+
+```rust
+struct MyType {
+    a: ComponentA,
+    b: ComponentB,
+}
+
+impl ExtractInstance for MyType {
+    type Query = (Read<ComponentA>, Read<ComponentB>);
+    type Filter = ();
+
+    fn extract((a, b): QueryItem<'_, Self::Query>) -> Option<Self> {
+        Some(MyType {
+          a: a.clone(),
+          b: b.clone(),
+        })
+    }
+}
+
+app.add_plugins(ExtractInstancesPlugin::<MyType>::extract_visible());
+```
+
+### Sprite Instancing
+
+Sprites were being rendered by generating a vertex buffer containing 4 vertices per sprite with position, UV, and possibly color data. This has proven to be very effective. However, having to split batches of sprites into multiple draws because they use a different color is suboptimal.
+
+Sprite rendering now uses an instance-rate vertex buffer to store the per-instance data. Instance-rate vertex buffers are stepped when the instance index changes, rather than when the vertex index changes. The new buffer contains an affine transformation matrix that enables translation, scaling, and rotation in one transform. It contains per-instance color, and UV offset and scale.
+
+This retains all the functionality of the previous method, enables the additional flexibility of any sprite being able to have a color tint and all still be drawn in the same batch, and uses a total of 80 bytes per sprite, versus 144 bytes previously. The practical result is a performance improvement of up to 40% versus the previous method!
+
+### Overall Performance vs 0.11
+
+3D Meshes
+
+2D Meshes
+
+Sprites
+
+UI
+
+### What's next for rendering performance?
```

Comment:
Imo we should mention parallel draw encoding, which I have a PR open for. We're just waiting for wgpu to land arcanization.

Reply:
I've slimmed down these "whats next" sections a lot. I think we should just cover this whenever we've finished it (or when we've kicked off the effort).